File: 2021/day01.rb
input = File.readlines("data/day01.txt", chomp: true).map { |e| e.to_i }

# Part 1

# larger = 0

# for i in 1..input.size
#     e = input[i].to_i
#     l = input[i-1].to_i
#     larger += 1 if e > l
# end

# puts larger

puts input.map.with_index { |e,i| e > input[i-1] }.count(true)

# Part 2

puts input.map.with_index { |e,i| e > input[i-3] }.count(true)

# larger = 0

# for i in 3..input.size
#     e = input[i].to_i
#     c = input[i-3].to_i
#     larger += 1 if e > c
# end

# puts larger

File: 2019/Python/day01.py
with open("data/day01.txt") as file:
    masses = [ int(n) for n in file.readlines() ]

def counter_upper(mass):
    return int(mass / 3) - 2

print(f"Part one = {sum(map(counter_upper, masses))}")

def double_checker(mass):
    extra_mass = fuel_required = counter_upper(mass)
    while counter_upper(extra_mass) > 0:
        # Oh noes I'm doing it twice!
        extra_mass = counter_upper(extra_mass)
        fuel_required += extra_mass
    return fuel_required

print(f"Part two = {sum(map(double_checker, masses))}")